### PR TITLE
OpenStack fix security groups (bsc#1146894)

### PR DESCRIPTION
**Pull request by flavio**
## Why is this PR needed?

Despite working, the OpenStack terraform deployment has some issues when it comes to security groups:
  * It exposes ports that are no longer needed by V4
  * It doesn't exposes ports that are needed
  * The security rules are too lax: they do not make any distinction between internal and external traffic
  * They are expressed using a deprecated resource of OpenStack

## What does this PR do?

This PR addresses all the issues I mentioned above:
  * Security groups are updated to reflect the contents of [our documentation](https://susedoc.github.io/doc-caasp/master/caasp-quickstart/single-html/#_ports) (BTW: the table just got updated inside of [this PR](https://github.com/SUSE/doc-caasp/pull/398)).
  * Use `networking_secgroup_rule_v2` resources to express rules are recommended by [upstream documentation](https://www.terraform.io/docs/providers/openstack/r/networking_secgroup_rule_v2.html).
  * Drop the security group for worker nodes: master and worker nodes with v4 are almost the same. The master nodes are like worker nodes with just the control plane services running on top of them.

## Anything else a reviewer needs to know?

Reviewing the whole diff at the same time can be a bit confusing. I highly recommend to go over each individual commit: they are well scoped and have nice explanation messages. They will make your life easier.

This PR oversees [this one from rafa](https://github.com/SUSE/skuba/pull/414).

## Info for QA

Nothing really changes to the end users:

* Deploy a cluster using our guidelines
* Ensure everything is working as expected

## Docs

No changes are needed inside of the documentation

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
We need a bug in bugzilla so we can make sure we test this during GMC validation. @flavio could you create a bug?

**Comment by flavio** [comment]
> We need a bug in bugzilla so we can make sure we test this during GMC validation. @flavio could you create a bug?

I've created https://bugzilla.suse.com/show_bug.cgi?id=1146894

I guess I have to include the bsc into the commits. Am I right?

**Comment by jordimassaguerpla** [comment]
> > We need a bug in bugzilla so we can make sure we test this during GMC validation. @flavio could you create a bug?
> 
> I've created https://bugzilla.suse.com/show_bug.cgi?id=1146894
> 
> I guess I have to include the bsc into the commits. Am I right?

in the title of the commits, so you will need to rebase.

**Comment by flavio** [comment]
I updated the commits to include the reference to the bugzilla entry

**Comment by NunoFilipeSantos** [comment]
@flavio Please review the commit message (it is too big) as it is failing CI. :)

File: ci/infra/openstack/registration.auto.tfvars [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 # SUSE CaaSP Product Registration Code
 #caasp_registry_code = ""
+caasp_registry_code = "INTERNAL-USE-ONLY-6e37-94d8"
```
[the remark this answers is not in the document]
Comment (flavio) [reply ci/infra/openstack/registration.auto.tfvars]: ouch, thanks!

File: ci/infra/openstack/security-groups-load-balancer.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  direction         = "ingress"
+  ethertype         = "IPv4"
+  protocol          = "tcp"
+  port_range_min    = 6443
+  port_range_max    = 6443
+  remote_ip_prefix  = "0.0.0.0/0"
+  security_group_id = "${openstack_networking_secgroup_v2.load_balancer.id}"
+}
+
+# Range of ports used by kubernetes when allocating services of type `NodePort`
```
Comment (lcavajani) [inline ci/infra/openstack/security-groups-load-balancer.tf]: NodePort are open directly the nodes so there is no need to add them on the LB so having this in the sg `common` is enough IMO.

By default we need TCP `32000` for dex and TCP `32001` for Gangway.

File: ci/infra/openstack/registration.auto.tfvars (hunk 1 above)
[the remark this answers is not in the document]
Comment (flavio) [reply ci/infra/openstack/registration.auto.tfvars]: Note well: the code has been deactivated by the SCC team

**Comment by flavio** [comment]
I pushed a fix to address @lcavajani's comment

File: ci/infra/openstack/security-groups-load-balancer.tf (hunk 2 above)
Comment (flavio) [reply ci/infra/openstack/security-groups-load-balancer.tf]: You're right. I fixed that

**Comment by flavio** [comment]
**do not merge:** it looks like gangway is not accessible now. Investigating...

**Pull request merged**